histolab/stain_normalizer.py

```diff
@@ -318,3 +318,104 @@ def _ordered_stack(stain_matrix: np.ndarray, order: List[int]) -> np.ndarray:
             return np.stack([stain_matrix[..., j] for j in order], -1)
 
         return _ordered_stack(stain_matrix, _get_channel_order(stain_matrix))
+
+
+class ReinhardStainNormalizer:
+    """Stain normalizer using the method of E. Reinhard et al. [1]_
+
+    References
+    ----------
+    .. [1] Reinhard, Erik, et al. "Color transfer between images." IEEE Computer
+        graphics and applications 21.5 (2001): 34-41.
+
+    """
+
+    def __init__(self):
+        self.target_means = None
+        self.target_stds = None
+
+    def fit(self, target_rgb: PIL.Image.Image) -> None:
+        """Fit stain normalizer using ``target_img``.
+
+        Parameters
+        ----------
+        target_rgb : PIL.Image.Image
+            Target image for stain normalization. Can be either RGB or RGBA.
+        """
+        means, stds = self.mean_std(target_rgb)
+        self.target_means = means
+        self.target_stds = stds
+
+    def transform(self, img_rgb: PIL.Image.Image) -> PIL.Image.Image:
+        """Normalize staining of ``img_rgb``.
+
+        Parameters
+        ----------
+        img_rgb : PIL.Image.Image
+            Image to normalize. Can be either RGB or RGBA.
+
+        Returns
+        -------
+        PIL.Image.Image
+            Image with normalized stain.
+        """
+        means, stds = self.mean_std(img_rgb)
+        img_lab = RgbToLab()(img_rgb)
+
+        mask = self._tissue_mask(img_rgb)
+        masked_img_lab = np.ma.masked_array(img_lab, ~mask)
+
+        norm_lab = (
+            ((masked_img_lab - means) * (self.target_stds / stds)) + self.target_means
+        ).data
+
+        for i in range(3):
+            original = img_lab[:, :, i].copy()
+            new = norm_lab[:, :, i].copy()
+            original[np.not_equal(~mask[:, :, 0], True)] = 0
+            new[~mask[:, :, 0]] = 0
+            norm_lab[:, :, i] = new + original
+
+        norm_rgb = LabToRgb()(norm_lab)
+        return norm_rgb
+
+    def mean_std(self, img_rgb: PIL.Image.Image) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
this should by private and I'd change the name to `_summary_measures`

Reply:
I'd call it `_summary_statistics`